doc/rados/operations/pg-repair.rst

```diff
@@ -1,4 +1,65 @@
+============================
 Repairing PG inconsistencies
 ============================
+Sometimes a placement group might become "inconsistent". To return the
+placement group to an active+clean state, you must first determine which 
+of the placement groups has become inconsistent and then run the "pg
+repair" command on it. This page contains commands for diagnosing placement
+groups and the command for repairing placement groups that have become
+inconsistent.
+
+.. highlight:: console
+
+Commands for Diagnosing Placement-group Problems
+================================================
+The commands in this section provide various ways of diagnosing broken placement groups.
+
+The following command provides a high-level (low detail) overview of the health of the ceph cluster::
+
+   # ceph health detail
+
+The following command provides more detail on the status of the placement groups::
+
+   # ceph pg dump --format=json-pretty
+
+The following command lists inconsistent placement groups::
+
+   # rados list-inconsistent-pg {pgid} 
+
+The following command lists inconsistent rados objects::
+
+   # rados list-inconsistent-obj {pgid} 
+ 
+The following command lists inconsistent snapsets in the given placement group::
+
+   # rados list-inconsistent-snapset {pgid}
+
+
+Commands for Repairing Placement Groups
+=======================================
+The form of the command to repair a broken placement group is::
+
+   # ceph pg repair {pgid}
+  
+Where "{pgid}" is the id of the affected placement group.
+
+For example::
+
+   # ceph pg repair 1.4
+
+More Information on Placement Group Repair
+==========================================
+Ceph stores and updates the checksums of objects stored in the cluster. When a scrub is performed on a placement group, the OSD attempts to choose an authoritative copy from among its replicas. Among all of the possible cases, only one case is consistent. After a deep scrub, Ceph calculates the checksum of an object read from the disk and compares it to the checksum previously recorded. If the current checksum and the previously recorded checksums do not match, that is an inconsistency. In the case of replica pools, any mismatch between the checksum of any replica of an object and the checksum of the authoritative copy means that there is an inconsistency.
```

Comment:
```suggestion
Ceph stores and updates the checksums of objects stored in the cluster. When a scrub is performed on a placement group, the OSD attempts to choose an authoritative copy from among its replicas. Among all of the possible cases, only one case is consistent. After a deep scrub, Ceph calculates the checksum of an object read from the disk and compares it to the checksum previously recorded. If the current checksum and the previously recorded checksums do not match, that is an inconsistency. In the case of replicated pools, any mismatch between the checksum of any replica of an object and the checksum of the authoritative copy means that there is an inconsistency.
```

I am not sure what you are trying to say here: "Among all of the possible cases, only one case is consistent"

Reply:
it's indeed a little confusing. but i think it means

> there is only one truth

Reply:
I'm attempting to refer to the single case that Ceph regards as consistent, and to show it as distinct from all the other cases. This single case is what Kefu is referring to as "truth". I could use language that is more like the language of set theory, but I think that would be even more obscure and confusing to most readers.